evalml/model_understanding/metrics.py

```diff
@@ -282,27 +258,30 @@ def roc_curve(y_true, y_pred_proba):
                   * `auc_score`: The area under the ROC curve.
     """
     y_true_ww = infer_feature_types(y_true)
-    y_true_np = _convert_ww_series_to_np_array(y_true_ww)
-    y_pred_proba = infer_feature_types(y_pred_proba).to_numpy()
-
-    if len(y_pred_proba.shape) == 1:
-        y_pred_proba = y_pred_proba.reshape(-1, 1)
```

Comment:
The `roc_curve` changes are so drastic to allow us to pass in nullable types, because we were manipulating `y_true` and `y_predict_proba` as numpy arrays instead of pandas series/dataframes.

We generally tend to stay in pandas in evalml, so I don't think it's so problematic to stay in pandas to avoid the extra nullable types logic. Though worth noting a slight increase in runtime when we switch to using pandas (the difference was proportional as I increased number of rows and whether or not the dtype was `int64` or `Int64`). Right now, I'm thinking that this is worth being able to remove the nullable type logic.

![image](https://user-images.githubusercontent.com/64278226/224779385-d3468e70-a3f8-476a-af2f-f0a913171b70.png)

The reason we can't stay in numpy is that a `series.to_numpy()` call from pandas nullable dtypes will always result in data with the `object` dtype, and this is not something numpy is going to change. Previously, we couldn't pass in the nullable types to the `label_binarizer` or `sklearn_roc_curve`, so moving to numpy right away via `_convert_ww_series_to_np_array` made sense, but now there's no need. But now that we can pass in pandas data with nullable dtypes, we should follow the rest of evalml and stay in pandas when possible.